jsettlers.graphics/src/main/java/jsettlers/graphics/map/MapContent.java

```diff
@@ -471,10 +470,12 @@ private void drawMain(FloatRectangle screen) {
 
 		if (placementBuilding != null) {
 			ShortPoint2D underMouse = this.context.getPositionOnScreen((float) mousePosition.getX(), (float) mousePosition.getY());
-			IMapObject mapObject = context.getMap().getMapObjectsAt(underMouse.x, underMouse.y);
+			if (0 <= underMouse.x && underMouse.x < width && 0 <= underMouse.y && underMouse.y < height) {
```

Comment:
contains point checks if a screen position is on the UI (regarding the javadoc and judging by its usage).